### ISSUE TITLE
quarto CLI does not throw an error code when failing which breaks some tests and feature 

### ISSUE THREAD

**Issue by cderv** [quarto-dev/quarto-r#198]
Reported at 
- https://github.com/quarto-dev/quarto-cli/issues/10295

````
══ Failed tests ════════════════════════════════════════════════════════════════
── Failure ('test-quarto.R:9:3'): quarto_run gives guidance in error ───────────
`quarto_run(c("rend", "--quiet"))` did not throw the expected error.
````

**Comment by cderv** [comment]
Should be solved with next  patch release, and next pre-release

**Comment by cderv** [comment]
This is fixed by 1.5.55 and latest 1.6 pre-release.
